src/txn/txn.c

```diff
@@ -1839,8 +1915,10 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
          * time we release our transaction so that the LSN any checkpoint gets will always reflect
          * visible data.
          */
-        __wt_readlock(session, &txn_global->visibility_rwlock);
-        locked = true;
+        if (!locked) {
+            __wt_readlock(session, &txn_global->visibility_rwlock);
+            locked = true;
+        }
```

Comment:
This looks like a bug-fix rather than a mechanical change?  Is it in the original?

Reply:
No, this is not a bug fix. I need to do this because in the oligarch path, I may have locked it already.